packages/autorest.python/autorest/codegen/serializers/builder_serializer.py

```diff
@@ -547,11 +555,16 @@ def create_http_request(builder: RequestBuilderType) -> List[str]:
         return retval
 
     def serialize_headers(self, builder: RequestBuilderType) -> List[str]:
-        retval = ["# Construct headers"]
-        for parameter in builder.parameters.headers:
+        headers = [
+            h
+            for h in builder.parameters.headers
+            if not builder.has_form_data_body or h.wire_name.lower() != "content-type"
```

Comment:
for form data body, we relay on core to set content-type, right?

Reply:
not core, but `requests` or `aiohttp`